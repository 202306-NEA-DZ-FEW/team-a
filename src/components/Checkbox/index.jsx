import { useTranslation } from "next-i18next";

function Checkbox({
  name,
  label,
  value,
  handleChange,
  handleBlur,
  error,
  touched,
}) {
  const { i18n } = useTranslation();
  return (
    <div>
      <div className='flex items-center gap-2'>
        <input
          type='checkbox'
          id={name}
          name={name}
          value={value}
          onChange={handleChange}
          onBlur={handleBlur}
          className={`checkbox checkbox-primary ${
            i18n?.language == "ar" ? "scale-x-[-1]" : ""
          }`}
        />
        <label htmlFor={name} className='label label-text'>
          {label}
        </label>
      </div>
      {/* <label className='label'>
        <span className='label-text-alt'></span>
        <span
          className={`label-text-alt ${touched && error ? "text-error" : ""}`}
        >
          {touched && error ? error : ""}
        </span>
      </label> */}
    </div>
  );
}

export default Checkbox;
