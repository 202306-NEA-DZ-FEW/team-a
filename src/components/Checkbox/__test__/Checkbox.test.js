import renderer from "react-test-renderer";

import Checkbox from "..";

it("renders correctly", () => {
  const tree = renderer.create(<Checkbox />).toJSON();
  expect(tree).toMatchSnapshot();
});
