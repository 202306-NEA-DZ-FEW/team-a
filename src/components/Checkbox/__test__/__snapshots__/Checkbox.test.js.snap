// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="flex items-center gap-2 pl-3"
  >
    <input
      className="checkbox checkbox-primary "
      type="checkbox"
    />
    <label
      className="label label-text"
    />
  </div>
</div>
`;
